src/WebJobs.Extensions.DurableTask/Correlation/DurableTaskCorrelationTelemetryInitializer.cs

```diff
@@ -0,0 +1,324 @@
+            }
+
+            if (initializeFromCurrent)
+            {
+                opTelemetry.Id = !string.IsNullOrEmpty(opTelemetry.Id) ? opTelemetry.Id : context.TelemetryId;
+                telemetry.Context.Operation.ParentId = !string.IsNullOrEmpty(telemetry.Context.Operation.ParentId) ? telemetry.Context.Operation.ParentId : context.TelemetryContextOperationParentId;
+            }
+            else
+            {
+                telemetry.Context.Operation.Id = !string.IsNullOrEmpty(telemetry.Context.Operation.Id) ? telemetry.Context.Operation.Id : context.TelemetryContextOperationId;
+                if (telemetry is ExceptionTelemetry)
+                {
+                    telemetry.Context.Operation.ParentId = context.TelemetryId;
+                }
+                else
+                {
+                    telemetry.Context.Operation.ParentId = !string.IsNullOrEmpty(telemetry.Context.Operation.ParentId) ? telemetry.Context.Operation.ParentId : context.TelemetryContextOperationParentId;
+                }
+            }
+        }
+
+        internal static void UpdateTelemetryW3C(ITelemetry telemetry, W3CTraceContext context)
+        {
+            OperationTelemetry opTelemetry = telemetry as OperationTelemetry;
+
+            bool initializeFromCurrent = opTelemetry != null;
+
+            if (initializeFromCurrent)
+            {
+                initializeFromCurrent &= !(opTelemetry is DependencyTelemetry dependency &&
+                    dependency.Type == SqlRemoteDependencyType &&
+                    dependency.Context.GetInternalContext().SdkVersion
+                        .StartsWith(RddDiagnosticSourcePrefix, StringComparison.Ordinal));
+            }
+
+            if (!string.IsNullOrEmpty(context.TraceState))
+            {
+                opTelemetry.Properties["w3c_tracestate"] = context.TraceState;
+            }
+
+            TraceParent traceParent = context.TraceParent.ToTraceParent();
+
+            if (initializeFromCurrent)
+            {
+                if (string.IsNullOrEmpty(opTelemetry.Id))
+                {
+                    opTelemetry.Id = traceParent.SpanId;
+                }
+
+                if (string.IsNullOrEmpty(context.ParentSpanId))
+                {
+                    telemetry.Context.Operation.ParentId = telemetry.Context.Operation.Id;
+                }
+            }
+            else
+            {
+                if (telemetry.Context.Operation.Id == null)
+                {
+                    telemetry.Context.Operation.Id = traceParent.TraceId;
+                }
+
+                if (telemetry.Context.Operation.ParentId == null)
+                {
+                    telemetry.Context.Operation.ParentId = traceParent.SpanId;
+                }
+            }
+        }
+
+        internal void SuppressTelemetry(ITelemetry telemetry)
+        {
+            // TODO change the strategy.
```

Comment:
This TODO is very vague. I would either elaborate on what the new strategy should be in the comment, or open a GitHub issue and reference it in your TODO.

Reply:
Done